### PR TITLE
Added section select to schedule generate page

### PR DESCRIPTION
**Pull request by Ahnaf-Ahmed**
Changes
---
Added a selection for number of sections to be offered. Defaults to 1 with a max of 6

Testing
---
Manual testing by outputting to console log

Linked Issue
---
https://github.com/SENG499-Company-4/Frontend/issues/80

Additional Info
---
image

### CODE REVIEW

File: app/src/components/pages/ScheduleGenerate.tsx [hunk 1]
```diff
@@ -166,14 +174,60 @@ function ScheduleGenerate() {
...
+                          <Select
+                            labelId={className + '-sections-select-label'}
+                            id={className + '-sections-select'}
+                            defaultValue={1}
+                            label="Sections to be offered"
+                            onChange={(event) => {
+                              setSections({ ...sections, [className]: event.target.value as number });
+                            }}
+                          >
+                            {[1, 2, 3, 4, 5, 6].map((amount) => (
```
Comment (bmazerolle) [inline app/src/components/pages/ScheduleGenerate.tsx:217]: why 6?

**Review by bmazerolle — CHANGES_REQUESTED** [review]
The code itself looks good but I think we should work on the UI a bit - it'll get very big and ugly quickly if Rich adds a lot of courses. Try to add the "sections" dropdown selector inline with the the classes that are selected, in the search bar.

Basically I would get rid of the grid you added, and just make each selected course in search bar larger, and containing the section count selector inline in that course object.

**Comment by Ahnaf-Ahmed** [comment]
> The code itself looks good but I think we should work on the UI a bit - it'll get very big and ugly quickly if Rich adds a lot of courses. Try to add the "sections" dropdown selector inline with the the classes that are selected, in the search bar.
> 
> Basically I would get rid of the grid you added, and just make each selected course in search bar larger, and containing the section count selector inline in that course object.

Just to be clear, in the chips in the search bar? 
image

If that's the case I feel like those are really small and the selection should at least be below the search bar. Even if said chips are made larger, I think we'll have a similar problem of it getting big and ugly with a lot of courses added in but now with a very large search bar instead of separate part of the page.

File: app/src/components/pages/ScheduleGenerate.tsx (hunk 1 above)
Comment (bmazerolle) [reply app/src/components/pages/ScheduleGenerate.tsx:224]: Ok I would change to 10 just to be safe, no harm in a bigger number

**Review by bmazerolle — CHANGES_REQUESTED** [review]
Nice job. I have two requested changes:
- Can you reset the course list after submission? Ideally the whole page. Could either clear the list or reload the page.
- Increase the max to 10, I think that will be big but safer than 66 since classes keep getting bigger

**Comment by Ahnaf-Ahmed** [comment]
> Nice job. I have two requested changes:
> 
> * Can you reset the course list after submission? Ideally the whole page. Could either clear the list or reload the page.
> * Increase the max to 10, I think that will be big but safer than 66 since classes keep getting bigger

Page now refreshes upon clicking OKin the dialog box 
image
 and section max is now 10

File: app/src/components/pages/ScheduleGenerate.tsx [hunk 2]
```diff
@@ -166,14 +177,78 @@ function ScheduleGenerate() {
...
+                            id={className + '-sections-select'}
+                            defaultValue={0}
+                            label="Sections to be offered"
+                            onChange={(event) => {
+                              setSections({ ...sections, [className]: event.target.value as number });
+                            }}
+                          >
+                            {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((amount) => (
+                              <MenuItem key={amount} value={amount}>
+                                {amount != 0 ? amount : 'Algorithm Determined'}
```
Comment (bmazerolle) [inline app/src/components/pages/ScheduleGenerate.tsx:222]: Just to have an option for not setting the count

**Review by bmazerolle — APPROVED** [review]

**Pull request merged**